src/Sarif/Schemata/sarif-schema.json

```diff
@@ -1628,6 +1679,8 @@
           "description": "An array of logical locations such as namespaces, types or functions.",
           "type": "array",
           "minItems": 0,
+          "uniqueItems": false,
```

Comment:
>false [](start = 25, length = 5)

[https://github.com/oasis-tcs/sarif-spec/issues/304](https://github.com/oasis-tcs/sarif-spec/issues/304). let's definitely mark this issue as one that needs resolving. I don't understand why we aren't requiring each logical location entry to be unique. There should only be one entry in the array for each presumptive logical location. Allowing duplication will be problematic in doing things like rewriting indices after results matching.

btw - one idea I had for tracking spot changes of this kind that the TC hasn't review is to create a PR for them and to leave them open in the SDK. For this one, we'd simply create a branch off develop, adjust uniqueItems to 'true' and leave it as a TODO. the PR would need to be built out in some cases with test fixes, etc., once the TC approves.

Reply:
It was `false` in the table in Issue #270, but I agree, I think we got that one wrong. The poster child for "non-unique" logical locations is ones that differ by `kind`, so they really _should_ be unique.

---
In reply to: [245125272](https://github.com/Microsoft/sarif-sdk/pull/1201#discussion_r245125272) [](ancestors = 245125272)

Reply:
NOT changing this because, as you say, the TC approved `false` here.

---
In reply to: [245137305](https://github.com/Microsoft/sarif-sdk/pull/1201#discussion_r245137305) [](ancestors = 245137305,245125272)